### PR TITLE
feat: Hide "Invite More People" button for Personal users - WPB-14654

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issue

Invite More People" button for Personal users is a legacy function and we can remove it.

**Reasons:**
- Phone number login has been disabled (access to the address book is not relevant)
- This is a relic from the past
- Do not want the app to make users send message via phone number

### Testing

Login with a personal account -> click on "+" button
---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

**Review by samwyndham — APPROVED** [review]
Nice! 🧹

File: wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift [hunk 1]
```diff
@@ -26,41 +26,37 @@ public struct SearchOptions: OptionSet {
...
     public static let contacts = SearchOptions(rawValue: 1 << 0)
 
-    /// Users found in your address book.
-
-    public static let addressBook = SearchOptions(rawValue: 1 << 1)
-
     /// Users which are a member of the same team as you.
 
-    public static let teamMembers = SearchOptions(rawValue: 1 << 2)
+    public static let teamMembers = SearchOptions(rawValue: 1 << 1)
```
Comment (netbe) [inline wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift:31]: is it normal that the option change here?

**Review by netbe — APPROVED** [review]
looks good just left a question

File: wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift (hunk 1 above)
Comment (samwyndham) [reply wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift:31]: I was thinking the same. My thought was that if it is not persisted anywhere it should be fine. But I didn't check the project that thoroughly

Comment (KaterinaWire) [reply wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift:31]: Yes, we don't persist it.
I can refactor it and create an Enum for `SearchOption`.

```
public enum SearchOption {
    case contacts
    case teamMembers
    case excludeNonActiveTeamMembers
    case excludeNonActivePartners
    case directory
    case conversations
    case services
    case federated
    case localResultsOnly
}
```
And we will pass around `searchOptions: [SearchOption]`
Maybe in the next PR, so it's more clear
@netbe and @samwyndham  what do you think?

Comment (netbe) [reply wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift:31]: I think it was good to check this was not persisted. For now let's keep it like this, I don't have a strong opinion on changing it to enum

Comment (samwyndham) [reply wire-ios-sync-engine/Source/UserSession/Search/SearchRequest.swift:31]: I think it is good to keep it as it is. An `OptionSet` seems a good solution to it's purpose.

**Pull request merged**